src/java/htsjdk/variant/variantcontext/GenotypeLikelihoods.java

```diff
@@ -72,6 +72,13 @@
     */
     private final static GenotypeLikelihoodsAllelePair[] PLIndexToAlleleIndex = calculatePLcache(MAX_ALT_ALLELES_THAT_CAN_BE_GENOTYPED);
 
+
+    /*
+    * a cache of the PL index to the anyploid alleles it represents over all possible numbers of alternate alleles
+    */
+    protected final static List<List<Integer>> plIndexToAlleleIndices = new ArrayList<List<Integer>>();
```

Comment:
How about keeping the data structure as is but lazily initializing it with the double checked lock pattern in `initializeAneuploidPLIndexToAlleleIndices(final int altAlleles, final int ploidy))`? `aneuploidPLIndexToAlleleIndices` will only exist for one ploidy and the operations will be thread safe.